_field-types/flat-object.md

```diff
@@ -0,0 +1,221 @@
+---
+layout: default
+title: Flat object
+nav_order: 43
+has_children: false
+parent: Object field types
+grand_parent: Supported field types
+---
+
+# Flat object field type
+
+In OpenSearch, you don't have to specify a mapping before indexing documents. If you don't specify a mapping, OpenSearch uses [dynamic mapping]({{site.url}}{{site.baseurl}}/field-types/mappings#dynamic-mapping) to map every field and its subfields in the document automatically. When you ingest documents such as logs, you may not know every field's subfield name and type in advance. In this case, dynamically mapping all new subfields can quickly lead to a "mapping explosion", where the growing number of fields may degrade the performance of your cluster. 
+
+Flat object solves this problem by not indexing the field itself or its subfields. Instead, for any JSON object, the values of all subfields and their paths are stored in two string fields. Subfields within the JSON are accessible using standard dot path notation but are not indexed for fast lookup.
```

Comment:
"Flat object solves this problem by treating the entire JSON object as string and does not indexing its subfields". The parent field is indexed, but the subfields are not indexable fields.